bot/exts/evergreen/branding.py

```diff
@@ -0,0 +1,552 @@
+import asyncio
+import itertools
+import json
+import logging
+import random
+import typing as t
+from datetime import datetime, time, timedelta
+from pathlib import Path
+
+import arrow
+import discord
+from discord.embeds import EmptyEmbed
+from discord.ext import commands
+
+from bot.bot import SeasonalBot
+from bot.constants import Branding, Colours, Emojis, MODERATION_ROLES, Tokens
+from bot.seasons import SeasonBase, get_all_seasons, get_current_season, get_season
+from bot.utils.decorators import with_role
+from bot.utils.exceptions import BrandingError
+from bot.utils.persist import make_persistent
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+STATUS_OK = 200  # HTTP status code
+
+FILE_BANNER = "banner.png"
+FILE_AVATAR = "avatar.png"
+SERVER_ICONS = "server_icons"
+
+BRANDING_URL = "https://api.github.com/repos/python-discord/branding/contents"
+
+PARAMS = {"ref": "seasonal-structure"}  # Target branch
+HEADERS = {"Accept": "application/vnd.github.v3+json"}  # Ensure we use API v3
+
+# A Github token is not necessary for the cog to operate,
+# unauthorized requests are however limited to 60 per hour
+if Tokens.github:
+    HEADERS["Authorization"] = f"token {Tokens.github}"
+
+
+class GithubFile(t.NamedTuple):
+    """
+    Represents a remote file on Github.
+
+    The sha hash is kept so that we can determine that a file has changed,
+    despite its filename remaining unchanged.
+    """
+
+    download_url: str
+    path: str
+    sha: str
+
+
+async def pretty_files(files: t.Iterable[GithubFile]) -> str:
+    """Provide a human-friendly representation of `files`."""
+    return "\n".join(file.path for file in files)
+
+
+async def time_until_midnight() -> timedelta:
```

Comment:
Adjusted in 466ce98 (`pretty_files` too).

Out of interest, where do you draw the line? I have a tendency to approach these as: if it can be async, then it should be. They're never used outside of an asynchronous context. Is there a point where the function is so simple that making it a coroutine is just not worth it?